### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by DmytroBondariev**
image
what is that?

### CODE REVIEW

**Review by vasylhnatiuk — CHANGES_REQUESTED** [review]
you didn't do it  `When the customer arrives at the shop his location should equal to shop location... After the shop he arrives home `

File: app/customer.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    product_cart: dict
+    home_location: list[int]
+    money: int
+    car: dict
+    money_spent: int | float = 0
+    current_location: list[int] = None
+
+    def choose_shop_or_stay_home(
+            self,
+            shops_list: list[Shop],
```
Comment (ArsenPidhoretskyi) [inline app/customer.py:18]:
```suggestion
            shops: list[Shop],

```
Why you need suffix?

File: app/customer.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        print(f"{self.name} has {self.money} dollars")
+        calculated_trips = dict()
+        for shop in shops_list:
+            calculated_trips[
+                self.calculate_trip_to_shop
+                (
+                    shop,
+                    fuel_price
+                )
+            ] = shop
```
Comment (ArsenPidhoretskyi) [inline app/customer.py:31]: In this case use dict-comprenhension

File: app/customer.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        y1 = self.home_location[1]
+        x2 = shop.location[0]
+        y2 = shop.location[1]
+        total_distance = round(
+            (((x2 - x1) ** 2 + (y2 - y1) ** 2) ** 0.5) * 2,
+            2
+        )
+        trip_fuel_consumption = \
+            total_distance * self.car["fuel_consumption"] \
+            / 100
```
Comment (ArsenPidhoretskyi) [inline app/customer.py:60]:
```suggestion
        trip_fuel_consumption = (
            total_distance * self.car["fuel_consumption"]
            / 100
        )

```

File: app/customer.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        fuel_expenses = trip_fuel_consumption * fuel_price
+        product_expenses = 0
+        for product, amount_to_buy in self.product_cart.items():
+            product_expenses += amount_to_buy * shop.products[product]
```
Comment (ArsenPidhoretskyi) [inline app/customer.py:64]: Use sum function

File: app/shop.py [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+        print(f"Thanks, {customer_name}, for your purchase!")
+        print("You have bought: ")
+        money_spent = 0
+        for product, amount in customer_product_cart.items():
+            money_spent += amount * self.products[product]
+            if amount == 1:
+                print(f"1 {product} for {self.products[product]} dollars")
+            print(
+                f"{amount} {product}s for "
+                f"{self.products[product] * amount} dollars")
```
Comment (ArsenPidhoretskyi) [inline app/shop.py:28]:
```suggestion
            plural = "s" if amount > 1 else ""
            print(f"{amount} {product}{plural} for {self.products[product] * amount} dollars")
```

**Review by ArsenPidhoretskyi — CHANGES_REQUESTED** [review]

**Review by ArsenPidhoretskyi — CHANGES_REQUESTED** [review]
All Remarks is not fixed

**Review by ArsenPidhoretskyi — APPROVED** [review]
Well done!

**Pull request closed without merging**